### PR TITLE
add turb_conv_affect_filter in callbacks

### PR DESCRIPTION
**Pull request by yairchn**
Adding here one of the EDMF filters into callbacks

### CODE REVIEW

File: examples/hybrid/callbacks.jl [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import ClimaCore.DataLayouts as DL
 import ClimaCore.Fields
 import ClimaComms
+import ClimaCore as CC
 import OrdinaryDiffEq as ODE
+const ca_dir = pkgdir(ClimaAtmos)
+import ClimaAtmos.Parameters as CAP
+include(joinpath(ca_dir, "tc_driver", "Surface.jl"))
```
Comment (charleskawczynski) [inline examples/hybrid/callbacks.jl:8]:
```suggestion
```
Need to qualify, `TCU.some_func_defined_in_surface()`

File: examples/hybrid/callbacks.jl [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import ClimaCore.DataLayouts as DL
 import ClimaCore.Fields
 import ClimaComms
+import ClimaCore as CC
 import OrdinaryDiffEq as ODE
+const ca_dir = pkgdir(ClimaAtmos)
```
Comment (charleskawczynski) [inline examples/hybrid/callbacks.jl:6]:
```suggestion
```

File: examples/hybrid/callbacks.jl [hunk 3]
```diff
@@ -80,6 +95,47 @@ function affect_filter!(integrator)
...
+    prog_face_column = CC.column(prog.f, inds...)
+    aux_cent_column = CC.column(aux.cent, inds...)
+    aux_face_column = CC.column(aux.face, inds...)
+    prog_column =
+        CC.Fields.FieldVector(cent = prog_cent_column, face = prog_face_column)
+    aux_column =
+        CC.Fields.FieldVector(cent = aux_cent_column, face = aux_face_column)
+
+    return TC.State(prog_column, aux_column, nothing)
+end
```
Comment (charleskawczynski) [inline examples/hybrid/callbacks.jl:138]: Don't redefine this.

File: examples/hybrid/callbacks.jl [hunk 4]
```diff
@@ -80,6 +95,47 @@ function affect_filter!(integrator)
...
+    (; edmf_cache, Δt) = p
+    (;
+        edmf,
+        param_set,
+        aux,
+        case,
+        surf_params,
+    ) = edmf_cache
+    t = integrator.t
+    Y = integrator.u
```
Comment (charleskawczynski) [inline examples/hybrid/callbacks.jl:108]:
```suggestion
    (; t, Y) = integrator
```

Comment (yairchn) [reply examples/hybrid/callbacks.jl]: @charleskawczynski I think this is not working:
```
LoadError: type ODEIntegrator has no field Y
--
  | Stacktrace:
  | [1] getproperty
  | @ ./Base.jl:42 [inlined]
  | [2] turb_conv_affect_filter!(integrator::O

```

you can see in the buildkite stack

Comment (charleskawczynski) [reply examples/hybrid/callbacks.jl:100]: Ah, yeah, just revert that

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
Bors r+

**Pull request merged**